### PR TITLE
Fixing issue between UpdatingVersion and UpdatingConfig at InPlace Up…

### PR DESCRIPTION
**Pull request by jparrill**
There was a bug into InPlace upgrade which mixes the UpdateConfig with UpdateVersion at the NodePool `status.conditions` field.

Now the NodePool shows properly the real conditions of the machines:

- When an Upgrade happens, it shows ConfigUpdate and VersionUpdate
- When the Config change happens (through MachineConfig) it shows only that the Config is updating.

I also changed the annotation message in the MachineSet because it always shows that a version change is happening even if that was not the case. Now shows that an upgrade is happening and shows the TargetConfig hash where it will upgrade to (that does not differentiate among Config or Version update) 

Signed-off-by: Juan Manuel Parrilla Madrid <jparrill@redhat.com>

**Which issue(s) this PR fixes**

- Fixes a Bug found at #[HOSTEDCP-403](https://issues.redhat.com/browse/HOSTEDCP-403)

### CODE REVIEW

**Comment by jparrill** [comment]
@jiezhao16 Still have an issue on the MachineSet annotation. It sets the VersionUpdate with the target version (Release) even if it's Config or Release.

The NodePool statuses are fixed in this Draft

File: hypershift-operator/controllers/nodepool/inplace.go [hunk 1]
```diff
@@ -171,6 +171,7 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
 	var status corev1.ConditionStatus
 	reason := ""
 	message := ""
+	status = "checking..."
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/inplace.go:174]: I'm not sure what this is coming from. Condition Status can either be true/false/unknown.

File: control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go [hunk 2]
```diff
@@ -211,7 +211,7 @@ func (r *Reconciler) reconcileInPlaceUpgrade(ctx context.Context, nodePoolUpgrad
 	// Signal in-place upgrade progress.
 	result, err := r.CreateOrUpdate(ctx, r.client, machineSet, func() error {
 		delete(machineSet.Annotations, nodePoolAnnotationUpgradeInProgressFalse)
-		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Updating version in progress. Target version: %q. Total Nodes: %d. Upgraded: %d", *machineSet.Spec.Template.Spec.Version, len(nodes), len(nodes)-nodeNeedUpgradeCount)
+		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Nodepool update in progress. Target Config hash: %s. Total Nodes: %d. Upgraded: %d", nodePoolUpgradeAPI.spec.targetConfigVersion, len(nodes), len(nodes)-nodeNeedUpgradeCount)
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go:214]: why this change? Also you mention config hash but you show ConfigVersion

File: hypershift-operator/controllers/nodepool/inplace.go (hunk 1 above)
Comment (jparrill) [reply hypershift-operator/controllers/nodepool/inplace.go:174]: maybe unknown it's better pick. Thanks

File: control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go (hunk 2 above)
Comment (jparrill) [reply control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go:214]: Because now, the message in the Annotation shows like a version upgrade even if it's just a Config one, so it's confusing for the user. 

In the other one, I will change from `config hash` to `config version`

File: hypershift-operator/controllers/nodepool/inplace.go [hunk 3]
```diff
@@ -201,7 +202,8 @@ func (r *NodePoolReconciler) reconcileMachineSet(ctx context.Context,
 		})
 	}
 
-	if message != "" && isUpdatingConfig {
+	if isUpdatingConfig {
```
Comment (enxebre) [inline hypershift-operator/controllers/nodepool/inplace.go:205]: why are we doing any different to the check above isUpdatingVersion here?

The message is being set here https://github.com/openshift/hypershift/pull/1978/files#diff-186fca173b913fab90bc6a26071e1255532f14ce6156a796e04de919d8e97c13R180

So even discriminate what is set in that annotation by config change / version change
Or just set a neutral message in that annotation and add a prefix here for both isUpdatingVersion  /  isUpdatingConfig

File: control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go [hunk 4]
```diff
@@ -211,7 +211,7 @@ func (r *Reconciler) reconcileInPlaceUpgrade(ctx context.Context, nodePoolUpgrad
 	// Signal in-place upgrade progress.
 	result, err := r.CreateOrUpdate(ctx, r.client, machineSet, func() error {
 		delete(machineSet.Annotations, nodePoolAnnotationUpgradeInProgressFalse)
-		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Updating version in progress. Target version: %q. Total Nodes: %d. Upgraded: %d", *machineSet.Spec.Template.Spec.Version, len(nodes), len(nodes)-nodeNeedUpgradeCount)
+		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Nodepool update in progress. Target Config version: %s. Total Nodes: %d. Upgraded: %d", nodePoolUpgradeAPI.spec.targetConfigVersion, len(nodes), len(nodes)-nodeNeedUpgradeCount)
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go:214]: Config version -> should be ConfigVersion hash, it's a (may be bad) name for the hash that contains both inputs.

File: control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go [hunk 5]
```diff
@@ -211,7 +211,7 @@ func (r *Reconciler) reconcileInPlaceUpgrade(ctx context.Context, nodePoolUpgrad
 	// Signal in-place upgrade progress.
 	result, err := r.CreateOrUpdate(ctx, r.client, machineSet, func() error {
 		delete(machineSet.Annotations, nodePoolAnnotationUpgradeInProgressFalse)
-		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Updating version in progress. Target version: %q. Total Nodes: %d. Upgraded: %d", *machineSet.Spec.Template.Spec.Version, len(nodes), len(nodes)-nodeNeedUpgradeCount)
+		machineSet.Annotations[nodePoolAnnotationUpgradeInProgressTrue] = fmt.Sprintf("Nodepool update in progress. Target Config version: %s. Total Nodes: %d. Upgraded: %d", targetConfigVersionHash, len(nodes), len(nodes)-nodeNeedUpgradeCount)
```
Comment (enxebre) [inline control-plane-operator/hostedclusterconfigoperator/controllers/inplaceupgrader/inplaceupgrader.go:214]: This should say target ConfigVersion hash
